opm/autodiff/StandardWellsDense.hpp

```diff
@@ -69,7 +72,9 @@ enum WellVariablePositions {
         template<typename FluidSystem, typename BlackoilIndices>
```

Comment:
these are almost equivalent in terms of the information accessible? you'll basically have to retrieve the type tag from there. alternatively, everything needed somewhere must be provided by BlackOilModel and it must be public. Also, BlackOilModel uses the property system to obtain these types anyway?

if you like this kind of fat interface design, I can accept it for code where I'm not the prime maintainer, but I'd strongly advice against it.